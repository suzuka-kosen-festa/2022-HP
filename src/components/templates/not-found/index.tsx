import type { FC } from "react";
import tw from "twin.macro";
import { Link } from "wouter";
import type { NotFoundProperties } from "./types/model";
import { StatusMessageSection } from "../../organisms/status-message-section";
import { Text } from "../../atoms/text";
import { Button } from "../../atoms/button";

const Main = tw.main`my-auto`;

const NotFound: FC<NotFoundProperties> = ({ status, message }) => (
  <Main aria-label="Not Found">
    <StatusMessageSection status={status} message={message}>
      <Text>このページは</Text>
      <Text>表示することができません。</Text>
      <Link to="/">
        <Button as="a" colorKey="yellow" role="button">
          トップページへ戻る
        </Button>
      </Link>
    </StatusMessageSection>
  </Main>
);

export { NotFound };
