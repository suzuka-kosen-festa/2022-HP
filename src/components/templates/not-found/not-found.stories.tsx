import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { NotFound } from ".";

type T = typeof NotFound;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  status: 404,
  message: "Not Found",
};

export default {
  args,
  argTypes: {},
  component: NotFound,
} as Meta;

export const Default: Story = {};
