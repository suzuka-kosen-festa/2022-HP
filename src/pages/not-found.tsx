import type { FC } from "react";
import { NotFound } from "../components/templates/not-found";

const NotFoundPage: FC = () => <NotFound status={404} message="Not Found" />;

export { NotFoundPage };
