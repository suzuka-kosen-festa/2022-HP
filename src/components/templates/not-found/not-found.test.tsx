import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./not-found.stories";

const { Default } = composeStories(stories);

const options = {
  name: "Not Found",
};

describe("(components) templates/not-found", () => {
  test("to be templates", () => {
    const { container } = render(<Default />);
    expect(container).toBeTemplate();
  });
  test("to be [role=main]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("main", options)).toBeInTheDocument();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
