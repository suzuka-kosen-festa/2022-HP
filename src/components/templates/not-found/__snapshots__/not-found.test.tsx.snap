// Vitest Snapshot v1

exports[`(components) templates/not-found > take snap shot 1`] = `
<div>
  <main
    aria-label="Not Found"
    class="css-1snmjsk"
  >
    <section
      aria-label="404"
      class="css-1dwazb3"
    >
      <div
        class="css-1y0riz5"
      >
        <div
          class="css-zo9kfe-StatusMessageSection"
        >
          <h1
            class="css-xv8iwh-StatusMessageSection"
          >
            404
          </h1>
        </div>
        <div
          class="css-jhqmv3"
        >
          <p
            class="css-15kij1l-StatusMessageSection"
          >
            Not Found
          </p>
          <div
            class="css-1tm5zuj-StatusMessageSection"
          >
            <p
              class="css-17zcuf2"
            >
              このページは
            </p>
            <p
              class="css-17zcuf2"
            >
              表示することができません。
            </p>
            <a
              class="css-jhmgok-Button"
              href="/"
              role="button"
              tabindex="0"
            >
              <p
                class="css-7bzdgk-Button"
              >
                トップページへ戻る
              </p>
            </a>
          </div>
        </div>
      </div>
    </section>
  </main>
</div>
`;
